py/client/pydeephaven/table.py

```diff
@@ -805,3 +805,67 @@ def delete(self, table: Table) -> None:
             self.session.input_table_service.delete(self, table)
         except Exception as e:
             raise DHError("delete data in the InputTable failed.") from e
+
+
+class MultiJoinInput:
+    """A MultiJoinInput represents the input tables, key columns and additional columns to be used in the multi-table
+    natural join.
+    """
+    table: Table
+    on: Union[str, Sequence[str]]
+    joins: Union[str, Sequence[str]] = None
+
+    def __init__(self, table: Table, on: Union[str, Sequence[str]], joins: Union[str, Sequence[str]] = None):
+        """Initializes a MultiJoinInput object.
+
+        Args:
+            table (Table): the right table to include in the join
+            on (Union[str, Sequence[str]]): the column(s) to match, can be a common name or an equality expression that
+                matches every input table, i.e. "col_a = col_b" to rename output column names.
+            joins (Union[str, Sequence[str]], optional): the column(s) to be added from the table to the result
+                table, can be renaming expressions, i.e. "new_col = col"; default is None
+        """
+        self.table = table
+        self.on = to_list(on)
+        self.joins = to_list(joins)
+
+
+def multi_join(input: Union[Table, Sequence[Table], MultiJoinInput, Sequence[MultiJoinInput]],
+               on: Union[str, Sequence[str]] = None) -> Table:
```

Comment:
The return type here is `Table` where the `deephaven` method returns `MultiJoinTable`.  `MultiJoinTable` does not inherit from `Table` and has a different API.  Users of `MultiJoinTable` must call `table()` to get a table to use.  I think I complained about this during that review.  Without adding a `MultiJoinTable` on the client side, the APIs are incompatible.
<img width="840" alt="image" src="https://github.com/user-attachments/assets/a002218b-b685-4e56-99da-4362e6332b3a">

Reply:
See https://github.com/deephaven/deephaven-core/pull/6020#discussion_r1765573954